lua/GammaAI/framework/mapping/MappingV2.lua

```diff
@@ -119,16 +150,44 @@ ConnectivityMatrix = Class({
     end,
 })
 
-MapArea = Class({
+---@class MapArea
+---@field xOffset number
+---@field zOffset number
+---@field maxi number
+---@field maxj number
+---@field compressLand boolean 
+---@field compressNavy boolean 
+---@field compressHover boolean 
+---@field compressAmph boolean
+---@field passableLand boolean | boolean[][]        # is a boolean if compressLand is true
+---@field passableNavy boolean | boolean[][]        # is a boolean if compressNavy is true
+---@field passableHover boolean | boolean [][]      # is a boolean if compressHover is true
+---@field passableAmph boolean | boolean[][]        # is a boolean if compressAmph is true
+---@field componentLand ConnectivityMatrix
```

Comment:
I'm a bit at a loss as to what this is. Sometimes it appears to be a number, sometimes a connectivity matrix but sometimes a matrix of either numbers or connectivity matrices.